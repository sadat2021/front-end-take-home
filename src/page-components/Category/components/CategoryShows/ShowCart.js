import { Flex, Box } from '@rebass/grid';
import Paragraph from 'shared-components/Typography/Paragraph';
import Header from 'shared-components/Typography/Header';
import PropTypes from 'prop-types';
import React from 'react';
import { StyledShowImage } from './styled';

function ShowCart({ show }) {
  return (
    <Box width={[1 / 2, 1 / 3, 1 / 4, 1 / 5]} mt={4} pr={2} pl={2}>
      <StyledShowImage src={show.images.squareLarge.url} alt={show.name} />
      <Header text={show.name} mt={2} linesToShow={1} />
      <Paragraph
        text={show.description}
        mt={2}
        linesToShow={3}
        variant="l"
        transparent
      />
    </Box>
  );
}

ShowCart.propTypes = {
  show: PropTypes.shape({
    name: PropTypes.string,
    description: PropTypes.string,
    images: PropTypes.shape({
      squareLarge: PropTypes.shape({
        url: PropTypes.string,
      }),
    }),
  }),
};

ShowCart.defaultProps = {
  show: {},
};

export default ShowCart;
