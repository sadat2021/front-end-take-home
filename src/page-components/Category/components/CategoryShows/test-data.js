export const shows = [
  {
    id: '0f6d4fe6-725e-4f43-add2-ac00007a96a6',
    slug: 'nsw-police-state-crime-command-investigations',
    name: 'NSW Police State Crime Command',
    description:
      'State Crime Command takes you into the minds of the investigators as they work to close active cases and bring justice for victims.',
    images: {
      squareLarge: {
        url: 'https://listnr-img-prod.scalabs.com.au/api/assets/mercury/a5be3cc8-3f76-4072-be3d-dd7c69d4260e/v0?width=320&height=320',
        pixelWidth: 320,
      },
    },
  },
  {
    id: 'e760f24d-6b25-4220-9376-ab21002bd3ec',
    slug: 'real-crime-australian-detectives',
    name: 'Real Crime Australian Detectives',
    description:
      'Real Crime investigative journalist Adam Shand speaks to Australian detectives about the iconic cases they have worked on, giving a step-by-step insider’s account of how the crimes were solved and the impact that these investigations had on the officers involved. \n\nFor more episodes download the free LiSTNR app.',
    images: {
      squareLarge: {
        url: 'https://listnr-img-prod.scalabs.com.au/api/assets/mercury/194a2766-6854-4682-9328-93e3a85e0a35/v1?width=320&height=320',
        pixelWidth: 320,
      },
    },
  },
  {
    id: '5dec054c-97ed-4104-92a9-aa4c003f9f05',
    slug: 'real-crime-feature',
    name: 'Real Crime Feature',
    description:
      'Real Crime investigative journalist Adam Shand takes you on an audio journey as he digs deep to uncover the untold story behind some of Australia’s most fascinating high-profile cases. Using a lifetime of experience, Adam will take you into his world where he finds the missing pieces and evidence that deliver Real Crime stories like you’ve never heard before.\n\nFor more episodes download the free LiSTNR app.',
    images: {
      squareLarge: {
        url: 'https://listnr-img-prod.scalabs.com.au/api/assets/mercury/840dffbc-0b2b-4c80-a0f7-5695ea92383d/v1?width=320&height=320',
        pixelWidth: 320,
      },
    },
  },
  {
    id: '4801919c-75e5-4537-adb2-aa4c003fa002',
    slug: 'real-crime-interview',
    name: 'Real Crime Interview',
    description:
      'Real Crime investigative journalist Adam Shand brings you into conversations with notorious and intriguing characters as he seeks to uncover the truth. The engaging human stories, from voices rarely heard in Australian media will amaze you.  \n\nFor more episodes download the free LiSTNR app',
    images: {
      squareLarge: {
        url: 'https://listnr-img-prod.scalabs.com.au/api/assets/mercury/54b06da0-7145-4aa3-bd07-0fec0286f5bd/v1?width=320&height=320',
        pixelWidth: 320,
      },
    },
  },
  {
    id: '13e374b6-5884-4d6b-aabc-ad7500396151',
    slug: 'true-crime-reporter',
    name: 'True Crime Reporter',
    description:
      'Investigative Reporter Robert Riggs created an original True Crime genre podcast based on three decades of real-life stories ripped from his reporter’s notebooks. \n\nWhen you hear the True Crime Reporter™, rest assured that Riggs was there inside the crime scene tape. \n\nIn every episode, Peabody Award-winning investigative reporter pulls out his reporter’s notebooks. His law enforcement sources open up their case files. \n\nThey sit down to talk. And you can listen to their Journey Into Darkness.',
    images: {
      squareLarge: {
        url: 'https://listnr-img-prod.scalabs.com.au/api/assets/mercury/ae6cbf7a-360f-43ea-9af8-b9a4744337fd/v1?width=320&height=320',
        pixelWidth: 320,
      },
    },
  },
  {
    id: 'c9e50fc4-db93-4557-a231-adf701853a52',
    slug: 'wheres-william-tyrrell',
    name: 'Wheres William Tyrrell',
    description:
      "On the 12th of September 2014, three-year-old William Tyrrell disappeared from the yard of his grandmother’s house in the quiet town of Kendall, New South Wales.\n\nA search and rescue mission soon became an investigation into his likely abduction, but who took him? Was the kidnapping planned or an opportunistic crime? Are we any closer to knowing what happened to William today than we were all those years ago?\n\n10 News First's Lia Harris and Natarsha Belling go through the entire story and strive to answer the question…Where’s William Tyrrell?",
    images: {
      squareLarge: {
        url: 'https://listnr-img-prod.scalabs.com.au/api/assets/mercury/bcc196e1-f3f5-4523-bf5c-8f170a3788a9/v3?width=320&height=320',
        pixelWidth: 320,
      },
    },
  },
];
