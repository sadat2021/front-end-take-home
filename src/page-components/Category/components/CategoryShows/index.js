import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { useSelector } from 'react-redux';
import { Flex } from '@rebass/grid';
import Paragraph from 'shared-components/Typography/Paragraph';
import Header from 'shared-components/Typography/Header';
import SortButton from 'shared-components/SortButton';
import { StyledBox, StyledCategoryShows, TextWrapper } from './styled';
import ShowCart from './ShowCart';

function CategoryShows({ shows, description }) {
  const [sortType, setSortType] = useState('ASC');
  const browser = useSelector((state) => state.device.browser);
  return browser === 'Chrome' ? (
    <StyledCategoryShows>
      <Flex
        justifyContent="space-between"
        alignItems="center"
        flexWrap="wrap"
        mt={4}
      >
        <StyledBox>
          {description && (
            <TextWrapper>
              <Paragraph text={description} variant="l" transparent />
            </TextWrapper>
          )}
        </StyledBox>
      </Flex>
      {shows && shows.length > 0 && (
        <>
          <Flex pl={2} justifyContent="space-between">
            <Header text={`${shows.length} Podcasts`} variant="xl" />
            <SortButton
              side="left"
              options={[
                { key: 'ASC', value: 'SORT A-Z' },
                { key: 'DESC', value: 'SORT Z-A' },
              ]}
              onOptionClick={(option) => {
                setSortType(option);
              }}
            />
          </Flex>
          <Flex flexWrap="wrap" flexDirection="row">
            {shows
              .sort((show1, show2) => {
                if (show1.name > show2.name) {
                  return sortType === 'ASC' ? 1 : -1;
                }
                return sortType === 'ASC' ? -1 : 1;
              })
              .map((show) => (
                <ShowCart key={show.id} show={show} />
              ))}
          </Flex>
        </>
      )}
    </StyledCategoryShows>
  ) : (
    <StyledCategoryShows>
      <Header variant="l" transparent>
        The page only work on the *latest Chrome version*
      </Header>
    </StyledCategoryShows>
  );
}

CategoryShows.propTypes = {
  shows: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string,
      description: PropTypes.string,
      images: PropTypes.shape({
        squareLarge: PropTypes.shape({
          url: PropTypes.string,
        }),
      }),
    })
  ),
  description: PropTypes.string,
};

CategoryShows.defaultProps = {
  shows: [],
  description: null,
};

export default CategoryShows;
