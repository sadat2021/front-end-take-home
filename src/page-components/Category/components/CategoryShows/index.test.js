import { render } from '@testing-library/react';
import CategoryShows from '.';
import { shows } from './test-data';

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useSelector: jest.fn().mockReturnValueOnce({
    device: {
      browser: 'Chrome',
    },
  }),
}));
describe('CategoryShows test', () => {
  test('Display the correct number of podcasts', () => {
    const { queryByText, container } = render(<CategoryShows shows={shows} />);
    expect(queryByText(`${shows.length} Podcasts`)).toBeInTheDocument;
    expect(container).toMatchSnapshot();
  });
  test('The Category shows page only work on the *latest Chrome version*', () => {
    jest.mock('react-redux', () => ({
      ...jest.requireActual('react-redux'),
      useSelector: jest.fn().mockReturnValueOnce({
        device: {
          browser: 'FireFox',
        },
      }),
    }));
    const { queryByText, container } = render(<CategoryShows shows={shows} />);
    expect(queryByText('The page only work on the *latest Chrome version*'))
      .toBeInTheDocument;
    expect(container).toMatchSnapshot();
  });
});
