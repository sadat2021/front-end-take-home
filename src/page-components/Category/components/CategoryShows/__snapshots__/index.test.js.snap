// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryShows test Display the correct number of podcasts 1`] = `
<div>
  <div
    class="src__Box-sc-1sbtrzs-0 sc-ieecCq kwrofL hSCNtY"
  >
    <h1
      class="sc-eCImPb sc-iCfMLu iLWCMF ealbgl"
    >
      The page only work on the *latest Chrome version*
    </h1>
  </div>
</div>
`;

exports[`CategoryShows test The Category shows page only work on the *latest Chrome version* 1`] = `
<div>
  <div
    class="src__Box-sc-1sbtrzs-0 sc-ieecCq kwrofL hSCNtY"
  >
    <h1
      class="sc-eCImPb sc-iCfMLu iLWCMF ealbgl"
    >
      The page only work on the *latest Chrome version*
    </h1>
  </div>
</div>
`;
